src/Driver.php

```diff
@@ -11,144 +11,172 @@ interface Driver
+     *
      * @return Promise<array|null>
      */
-    public function stat(string $path): Promise;
+    public function getStatus(string $path): Promise;
 
     /**
-     * Same as stat() except if the path is a link then the link's data is returned.
+     * Same as {@see Driver::getStatus()} except if the path is a link then the link's data is returned.
+     *
+     * If the requested path does not exist the resulting Promise will resolve to NULL.
      *
      * @param string $path The file system path to stat.
+     *
      * @return Promise<array|null> A promise resolving to an associative array upon successful resolution.
      */
-    public function lstat(string $path): Promise;
+    public function getLinkStatus(string $path): Promise;
 
     /**
      * Create a symlink $link pointing to the file/directory located at $target.
      *
      * @param string $target
      * @param string $link
+     *
      * @return Promise<void>
      */
-    public function symlink(string $target, string $link): Promise;
+    public function createSymlink(string $target, string $link): Promise;
 
     /**
      * Create a hard link $link pointing to the file/directory located at $target.
      *
      * @param string $target
      * @param string $link
+     *
      * @return Promise<void>
      */
-    public function link(string $target, string $link): Promise;
+    public function createHardlink(string $target, string $link): Promise;
 
     /**
-     * Read the symlink at $path.
+     * Resolve the symlink at $path.
      *
      * @param string $target
+     *
      * @return Promise<string>
      */
-    public function readlink(string $target): Promise;
+    public function resolveSymlink(string $target): Promise;
 
     /**
-     * Rename a file or directory.
+     * Move / rename a file or directory.
      *
      * @param string $from
      * @param string $to
+     *
      * @return Promise<void>
      */
-    public function rename(string $from, string $to): Promise;
+    public function move(string $from, string $to): Promise;
 
     /**
      * Delete a file.
      *
      * @param string $path
+     *
      * @return Promise<void>
      */
-    public function unlink(string $path): Promise;
+    public function deleteFile(string $path): Promise;
 
     /**
-     * Create a director.
+     * Create a directory.
      *
      * @param string $path
-     * @param int $mode
-     * @param bool $recursive
+     * @param int    $mode
+     *
      * @return Promise<void>
      */
-    public function mkdir(string $path, int $mode = 0777, bool $recursive = false): Promise;
+    public function createDirectory(string $path, int $mode = 0777): Promise;
+
+    /**
+     * Create a directory recursively.
+     *
+     * @param string $path
+     * @param int    $mode
+     *
+     * @return Promise<void>
+     */
+    public function createDirectories(string $path, int $mode = 0777): Promise;
 
     /**
```

Comment:
I think this could be a wrapper and leave the bool on `createDirectory()`. The name might be confusing as well, maybe `createDirectoryRecursive()` instead?